src/osd/PrimaryLogPG.cc

```diff
@@ -6706,14 +6706,15 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	        oi.size - op.extent.truncate_size);
 	      ctx->modified_ranges.union_of(trim);
 	      ctx->clean_regions.mark_data_region_dirty(op.extent.truncate_size, oi.size - op.extent.truncate_size);
+	      oi.clear_data_digest();
 	    }
 	    if (op.extent.truncate_size != oi.size) {
               truncate_update_size_and_usage(ctx->delta_stats,
                                              oi,
                                              op.extent.truncate_size);
+	      //truncate modify oi.size, need clear old data_digest and DIGEST flag
+	      oi.clear_data_digest();
```

Comment:
Oops, the commit title says " osd: Adjust code style" but this change isn't just a refactor – some extra in the git's stage area perhaps?

Anyway, I think they deserve a separate commit describing the intention.

Reply:
done, move the line to the other commit